plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibraries.java

```diff
@@ -65,28 +72,47 @@
   }
   
   /**
-   * @return the library with the specified ID, or null if not found
+   * Returns the library with the specified ID, or null if not found.
    */
   public static Library getLibrary(String id) {
-    // todo this method is only called by LibraryClasspathContainerResolverService and tests.
-    // Perhaps here or in LibraryClasspathContainerResolverService is where we should 
-    // resolve the transitive dependencies
     return libraries.get(id);
   }
+
+  private static final LoadingCache<IJavaProject, Library> masterLibraries =
+      CacheBuilder.newBuilder().build(new CacheLoader<IJavaProject, Library>() {
+
+        @Override
+        public Library load(IJavaProject project) throws JavaModelException {
```

Comment:
No one throws `JavaModelException`. If we remove it, we can just call `.getUnchekced()` in `getMasterLibrary()` below. It would be good to leave a comment here saying "Do not throw a checked exception, as cache clients call getUnchecked()", if we remove it.

Reply:
good catch, done